src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfoExtensions.cs

```diff
@@ -45,7 +50,25 @@ public static ProjectInfo CreateProjectInfo(this ProjectFileInfo projectFileInfo
                 language: LanguageNames.CSharp,
                 filePath: projectFileInfo.FilePath,
                 outputFilePath: projectFileInfo.TargetPath,
-                compilationOptions: projectFileInfo.CreateCompilationOptions());
+                compilationOptions: projectFileInfo.CreateCompilationOptions(),
+                analyzerReferences: analyzerReferences);
+        }
+
+        private static IEnumerable<AnalyzerReference> ResolveAnalyzerReferencesForProject(ProjectFileInfo projectFileInfo)
+        {
+            return projectFileInfo.Analyzers
+                .GroupBy(x => Path.GetDirectoryName(x))
+                .SelectMany(singleAnalyzerPackageGroup =>
+                {
+                    // Is there better way to figure out entry assembly for specific nuget analyzer package? And is there even entry assembly or is better way to just lookup all referenced assemblies?
+                    return singleAnalyzerPackageGroup
+                        .Where(x => x.EndsWith("Analyzers.dll") || x.EndsWith("Analyzer.dll"))
+                        .Select(analyzerMainAssembly => {
+                            var assemblyLoader = new AnalyzerAssemblyLoader();
```

Comment:
We're creating a new assembly loader with another `ConcurrentDictionary` for every directory that an analyzer is contained in? Can't this be shared? Or, better, maybe just use OmniSharp's `IAssemblyLoader`?

Reply:
I think i copied this way from roslyn code base, i will test if it work with shared loader. I add one method to assembly loader and make it implement IAnalyzerAssemblyLoader to reduce duplication.